### ISSUE TITLE
ASEditableTextNode contents not accessible to VoiceOver

### ISSUE THREAD

**Issue by ay8s** [TextureGroup/Texture#1135]
Just had a report of this, it looks like the contents typed into an ASEditableTextNode aren't accessible when using VoiceOver. Will try and dig in tomorrow and see if I can do a pull request.

**Comment by maicki** [comment]
@ay8s Hey, did you find something out yet? Thanks!

**Comment by ay8s** [comment]
For now I ended up updating the accessibility value with the text from the textView in editableTextNodeDidUpdateText: 

Also digging into a few other reports. Curious if Pinterest has any issues at all with ASEditableTextNode accessibility? Seems odd it's not free as a wrapper around UITextView.

**Comment by ay8s** [comment]
Digging into this one some more, there are pretty drastic differences in what VoiceOver reads between ASEditableTextNode and UITextView. In this example I simply put a ASEditableTextNode (green) at the top and a UITextView (red) below. Both are within a ASViewController with the UITextView wrapped in a ASDisplayNode.

No other settings were changed for either the ASEditableTextNode or the UITextView.

ASEditableTextNode
- Doesn't say that it is a text field.
- Doesn't say "Double tap to edit"
- Doesn't say how to access misspelled words.

When inputting characters...
- Doesn't read completed words. 
- Doesn't read insertion points. 

Will continue to dig in, now sure if there may be something conflicting with the expected VO output.

Video: https://cl.ly/49c21c61cd58

**Comment by ay8s** [comment]
Digging in more to this today after exploring just using UITextView. Seems that if I use ASDK_ACCESSIBILITY_DISABLE it works correctly. Oddly ASViewController's that use ASEditableTextNode as the root node work fine, only when a ASEditableTextNode is included as a subnode default behavior is lost.

**Comment by ay8s** [comment]
@maicki @nguyenhuy So I think I've figured out the cause. When a ASEditableTextNode is a root node its textView is available via "view". This isn't the case when it is a subnode. When we're collecting the accessibility elements here https://github.com/TextureGroup/Texture/blob/465abb1ded1777e7573a65e70df0ef0a2bda0df4/Source/Details/_ASDisplayViewAccessiblity.mm#L215-L234 we're not looking for the textView. 

I quickly hacked together adding the textView to the elements array and VoiceOver returned the correct information back to the user. 

Any downsides you can see adding the following?
```
  for (ASDisplayNode *subnode in node.subnodes) {
    if (subnode.isAccessibilityElement) {
      
      // An accessiblityElement can either be a UIView or a UIAccessibilityElement
      if (subnode.isLayerBacked) {
        // No view for layer backed nodes exist. It's necessary to create a UIAccessibilityElement that represents this node
        UIAccessibilityElement *accessiblityElement = [ASAccessibilityElement accessibilityElementWithContainer:view node:subnode containerNode:node];
        [elements addObject:accessiblityElement];
      } else {
        // Accessiblity element is not layer backed just add the view as accessibility element
        [elements addObject:subnode.view];
      }
    } else if (subnode.isLayerBacked) {
      // Go down the hierarchy of the layer backed subnode and collect all of the UIAccessibilityElement
      CollectUIAccessibilityElementsForNode(subnode, node, view, elements);
    } else if ([subnode accessibilityElementCount] > 0) {
      // UIView is itself a UIAccessibilityContainer just add it
      [elements addObject:subnode.view];
    } else if([subnode isKindOfClass:[ASEditableTextNode class]]){
      // UITextView is itself a UIAccessibilityElement just add it
      [elements addObject:((ASEditableTextNode *)subnode).textView];
    }
  }
```

**Comment by garrettmoon** [comment]
Could we set isAccessibilityElement to YES on init in ASEditableTextNode? Then it would add subside.view which is the textView in this case?

**Comment by ay8s** [comment]
Closing
